src/nagini_translation/lib/program_nodes.py

```diff
@@ -376,6 +376,26 @@ def __init__(self, name: str, superscope: PythonScope,
         self._has_classmethod = False
         self.type_vars = OrderedDict()
         self.definition_deps = set()
+        self.is_adt = name == 'ADT' # This flag can also be set transitively
```

Comment:
Could you add a sentence that says when this should be set? I.e., that it should be set both for the classes that define an ADT and for the classes that represent ADT constructors?

Reply:
Sure, done.